pkg/controller/vitesskeyspace/vitesskeyspace_controller.go

```diff
@@ -198,3 +216,18 @@ func (r *ReconcileVitessKeyspace) Reconcile(request reconcile.Request) (reconcil
 	reconcileCount.WithLabelValues(vtk.Labels[planetscalev2.ClusterLabel], vtk.Spec.Name, metrics.Result(err)).Inc()
 	return result, err
 }
+
+// newWrangler initializes a new Vitess Wrangler that gives us access to information
+// about resharding workflows.
+func newWrangler(ctx context.Context, lockserverSpec planetscalev2.VitessLockserverParams) (*wrangler.Wrangler, error) {
+	// We need to initialize for the first time if we got here.
+	ts, err := toposerver.Open(ctx, lockserverSpec)
```

Comment:
We need to ensure this gets closed somehow, or else we'll leak ref counts for this connection.